### PR TITLE
Add ASP.NET Core manual instrumentation sample

### PR DESCRIPTION
**Pull request by zacharycmontoya**
This PR adds an ASP.NET Core 3.1 application with handwritten middleware using the Datadog.Trace library to demonstrate how to instrument the incoming requests. This application also uses Serilog to demonstrate how to manually correlate logs and traces.

This sample adds three custom Datadog middleware classes to the ASP.NET Core pipeline to add visibility using the `Datadog.Trace` library:
1. `DatadogTracingRequestStartMiddleware` - This middleware must execute first in the ASP.NET Core request pipeline to create accurate Datadog spans, and is enabled by calling the extension method `UseDatadogTracingRequestStartMiddleware()`
1. `DatadogTracingExceptionLoggerMiddleware` - This middleware should execute early in the pipeline after the error handling middleware, and is enabled by calling the extension method `UseDatadogTracingExceptionLoggerMiddleware()`
1. `DatadogSerilogLogCorrelationMiddleware` - This middleware is optional and demonstrates how to add Datadog properties to the Serilog log context so that they will be emitted with the log events if the Serilog logger is enriched with the log context. It is enabled by calling the extension method `UseDatadogSerilogLogCorrelationMiddleware()`

README link: https://github.com/DataDog/dd-trace-dotnet/blob/zach/samples/aspnetcore/customer-samples/AspNetCoreWithSerilog/README.md

@DataDog/apm-dotnet

### CODE REVIEW

File: customer-samples/AspNetCoreWithSerilog/wwwroot/lib/bootstrap/dist/css/bootstrap-grid.css [hunk 1]
```diff
@@ -0,0 +1,3719 @@
+/*!
```
Comment (lucaspimentel) [inline customer-samples/AspNetCoreWithSerilog/wwwroot/lib/bootstrap/dist/css/bootstrap-grid.css:1]: I would like it if we remove all the extra static files and keep the js/css super simple. Alternatively, we should consider move "customer samples" into another repo altogether.

File: customer-samples/AspNetCoreWithSerilog/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+cd AspNetCoreWithSerilog
+dotnet run
+```
+
+## Results
+### Successful request
+![Datadog UI for a successful request with one ASP.NET Core span](https://user-images.githubusercontent.com/13769665/96036749-c2cb3280-0e19-11eb-9ff1-b9771778d032.PNG)
+
+### Failing request
+![Datadog UI for a failing request with one ASP.NET Core span](https://user-images.githubusercontent.com/13769665/96036754-c5c62300-0e19-11eb-9401-5796747b9ae3.PNG)
```
[the remark this answers is not in the document]
Comment (macrogreg) [reply customer-samples/AspNetCoreWithSerilog/README.md:24]: And later:

To see your logs make sure you have subscribed to this service, go to that page (link) click X and Y...

File: customer-samples/AspNetCoreWithSerilog/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+## Application setup
+To run this application, run the following commands:
+
+```
+cd AspNetCoreWithSerilog
+dotnet run
+```
+
+## Results
+### Successful request
```
[the remark this answers is not in the document]
Comment (macrogreg) [reply customer-samples/AspNetCoreWithSerilog/README.md:20]: Make sure not to expose an actual API key :)

**Review by macrogreg — CHANGES_REQUESTED** [review]
Thank you for doing this, Zach. This is great and very helpful for customers.

I would like to ask you to make a blog post out of it and place it into here: [https://github.com/DataDog/dd-trace-dotnet/tree/master/blog](https://github.com/DataDog/dd-trace-dotnet/tree/master/blog).
For that, we could modify your existing README file, copy it into the blog, and then have the actual readme simply linking to the blog post.

I have left PR comments on the readme file on what we should modify to make it a blog post. Overall those are a few minor things. The idea is that a user not overly familiar with the matter should be able to follow it.

I left some minor code comments as well.

Thanks!

**Comment by zacharycmontoya** [comment]
Closing due to inactivity and no current need for it

**Pull request closed without merging**